src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/WaitForBuildStepTest.java

```diff
@@ -112,7 +178,7 @@ private static FlowNode findFirstNodeWithDescriptor(FlowExecution execution, Cla
         return null;
     }
 
-    private WorkflowJob createWaitingDownStreamJob(Result result) throws Exception {
+    private WorkflowJob createWaitingDownStreamJob(Result result, boolean indefiniteWait) throws Exception {
         WorkflowJob ds = j.jenkins.createProject(WorkflowJob.class, "ds");
         ds.setDefinition(new CpsFlowDefinition(
             "import org.jenkinsci.plugins.workflow.support.steps.build.WaitForBuildAction\n" +
```

Comment:
BTW it would be more natural to use `semaphore` and then have test code unblock once some condition is satisfied.

Reply:
Agreed, but I can't think of a way to coordinate that between the test code and the pipeline implementation of the test jobs.